src/formula.jl

```diff
@@ -24,9 +24,138 @@ macro ~(lhs, rhs)
     return ex
 end
 
-#
-# TODO: implement contrast types in Formula/Terms
-#
+function Base.show(io::IO, f::Formula)
+    print(io,
+          string("Formula: ",
+                 f.lhs == nothing ? "" : f.lhs, " ~ ", f.rhs))
```

Comment:
You could put this all on one line and still be under the usual cutoff. Also I think it's typically preferred to use `===` when comparing against `nothing`.